src/components/dataset/data/DatasetDataView.jsx

```diff
@@ -0,0 +1,220 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import {
+  resetQuery,
+  runQuery,
+  getDatasetById,
+  getDatasetPolicy,
+  countResults,
+  getUserDatasetRoles,
+  pageQuery,
+} from 'actions/index';
+import { FilterList, Info, People } from '@material-ui/icons';
+
+import QueryView from 'components/common/query/QueryView';
+import LoadingSpinner from 'components/common/LoadingSpinner';
+import QueryViewSidebar from './sidebar/QueryViewSidebar';
+import InfoView from './sidebar/panels/InfoView';
+import ShareSnapshot from './sidebar/panels/ShareSnapshot';
+import {
+  DATAREPO_ROW_ID_COLUMN_NAME,
+  DATASET_INCLUDE_OPTIONS,
+  GOOGLE_CLOUD_RESOURCE,
+  RESOURCE_TYPE,
+} from '../../../constants';
+
+const QUERY_LIMIT = 1000;
+
+function DatasetQueryView({
```

Comment:
Did we decide to make all new components in Typescript? I can't remember. It would be cool to see Typescript proliferate in our codebase, though!

Reply:
Definitely - It was just more than I wanted to take on with this PR. I did attempt to rework any models to consistent var names/etc, which should help with that transition.